Bolts/Common/BFTask.m

```diff
@@ -33,24 +33,30 @@ @interface BFTask () {
 @property (atomic, assign, readwrite, getter=isCompleted) BOOL completed;
 
 @property (nonatomic, strong) NSObject *lock;
-@property (nonatomic, strong) NSCondition *condition;
+@property (nonatomic, assign) dispatch_semaphore_t completedSemaphore;
```

Comment:
Hmm, I feel this still needs to be `strong` when compiled on iOS 6.0+ / OS X 10.8+, as it will become a dangling pointer otherwise because it's managed by ARC.